src/shapes/Text/StyledText.ts

```diff
@@ -188,11 +188,21 @@ export abstract class StyledText<
       this._setLineStyle(lineIndex);
     }
 
-    if (!Object.keys(this._getStyleDeclaration(lineIndex, charIndex)).length) {
-      this._setStyleDeclaration(lineIndex, charIndex, {});
-    }
+    // first create a new object that is a merge of existing and new
+    const newStyle: TextStyleDeclaration = {
+      ...this._getStyleDeclaration(lineIndex, charIndex),
+      ...styles,
+    };
+
+    // then delete what is undefined in styles from newStyle
+    Object.keys(styles).forEach((key) => {
+      if (styles[key as keyof TextStyleDeclaration] === undefined) {
+        delete newStyle[key as keyof TextStyleDeclaration];
+      }
+    });
```

Comment:
Since you are creating a new object I think it is better to filter out undefined beforehand instead of creating it and then deleting stuff

```suggestion
    // first create a new object that is a merge of existing and new
    const newStyle: TextStyleDeclaration = {
      ...this._getStyleDeclaration(lineIndex, charIndex),
      ...Object.fromEntries(Object.entries(styles).filter(([key, value]) => value!==undefined),
    };

```

Reply:
can we also rename styles to style
very confusing, it is not an array, right?
Why don't we support an array?

Reply:
yes is a single style.
We don't support an array becuase this is a single style object that you apply to a range of text.

Reply:
This code is not equivalent. We need to delete what is in the old style as well, we are not cleaning up styles.

The equivalent would be something in the lines of:
- mixing the entries of styles and the previousStyle
- loop with a filter
- do the fromEntries

Reply:
I disagree, didn't you delete only the incoming `stylez arg?
Let's change this, see how confusing it is, this thread is evidence